robot-agent/mqtt.js

```diff
@@ -215,6 +215,74 @@ const staticInfo = () => {
   });
 };
 
+// list of self checks to run
+const selfChecks = [
+  {
+    name: 'unshare supported',
+    check: () => {
+      // check if unshare is available
+      try {
+        const result = execSync('unshare -rm whoami', {encoding: 'utf8'});
+        // check if it returns root
+        return result.trim() == 'root';
+      }
+      catch (e) {
+        return false;
+      }
+    },
+    error: 'unshare not supported, add kernel.apparmor_restrict_unprivileged_userns = 0 to /etc/sysctl.conf'
+  },
+  // check if bash is installed and is the default shell
+  {
+    name: 'bash installed',
+    check: () => {
+      // check if bash is available
+      try {
+        const result = execSync('which bash', {encoding: 'utf8'});
+        // check if it returns a path
+        return result.trim() != '';
+      }
+      catch (e) {
+        return false;
+      }
+    },
+    error: 'bash not installed, install bash'
+  },
+  // check if bash is the default shell
+  {
+    name: 'bash default shell',
+    check: () => {
+      // check if bash is the default shell
+      try {
+        const result = execSync('echo $SHELL', {encoding: 'utf8'});
+        return result.trim() == '/bin/bash';
+      }
+      catch (e) {
+        return false;
+      }
+    },
+    error: 'bash not default shell, set bash as default shell'
+  }
+]
+
+const executeSelfChecks = () => {
+  log.info('executing self checks');
+  data.update(`${AGENT_PREFIX}/status/selfChecks`, {running: true});
+  const errors = [];
+  selfChecks.forEach(check => {
+    if (!check.check()) {
+      errors.push(check.error);
+    }
+  });
+  if (errors.length > 0) {
+    log.error('self checks failed:', errors);
+    data.update(`${AGENT_PREFIX}/status/selfChecks`, {errors});
+  } else {
+    log.info('self checks passed');
+    data.update(`${AGENT_PREFIX}/status/selfChecks`, {success: true});
```

Comment:
As discussed, I think we should *not* send the error message, only the test result. The front-end should read the error message from a shared file based on the error ID or name. So what we send to the front-end should just be an object `{ [errorName]: errorResult, .. }`.

Reply:
For example: 
`{ 'unshare': 'cannot run unshare -rm, please do ...', condition: (result) => result.trim() == 'root', command: 'unshare -rm whoami'}`